raw

dmbavykin opened the pull request.

Commit:
few gems added

	new file:   .rspec
	modified:   Gemfile
	modified:   Gemfile.lock
	new file:   spec/rails_helper.rb
	new file:   spec/spec_helper.rb

Commit:
categories added

	new file:   app/models/category.rb
	new file:   db/migrate/20180127125052_create_categories.rb
	new file:   db/schema.rb
	new file:   spec/models/category_spec.rb

Commit:
authors added

Commit:
admin panel added

Commit:
books added

Commit:
shoulda-matchers added

Commit:
setuping views

Commit:
customer and rating added

Commit:
orders added

Commit:
implementing default styles

Commit:
home page and catalogs added

	modified:   Gemfile
	modified:   Gemfile.lock
	modified:   app/assets/stylesheets/style.scss
	new file:   app/controllers/books_controller.rb
	modified:   app/controllers/catalogs_controller.rb
	modified:   app/controllers/home_controller.rb
	modified:   app/helpers/application_helper.rb
	new file:   app/views/books/show.html.haml
	modified:   app/views/catalogs/index.html.haml
	modified:   app/views/home/home.html.haml
	modified:   app/views/layouts/main.html.haml
	modified:   config/routes.rb
	modified:   spec/factories/books.rb

Commit:
refactoring

Commit:
displaying reviews added

	modified:   app/controllers/catalogs_controller.rb
	modified:   app/helpers/application_helper.rb
	modified:   app/views/books/show.html.haml
	modified:   spec/factories/ratings.rb

Commit:
refactoring

Commit:
refactoring factories

Commit:
main page slider configured

Commit:
cloudinary for images added

Commit:
logic of working with images changed

Commit:
devise added

	modified:   Gemfile
	modified:   Gemfile.lock
	new file:   app/models/user.rb
	new file:   app/views/devise/confirmations/new.html.haml
	new file:   app/views/devise/mailer/confirmation_instructions.html.haml
	new file:   app/views/devise/mailer/email_changed.html.haml
	new file:   app/views/devise/mailer/password_change.html.haml
	new file:   app/views/devise/mailer/reset_password_instructions.html.haml
	new file:   app/views/devise/mailer/unlock_instructions.html.haml
	new file:   app/views/devise/passwords/edit.html.haml
	new file:   app/views/devise/passwords/new.html.haml
	new file:   app/views/devise/registrations/edit.html.haml
	new file:   app/views/devise/registrations/new.html.haml
	new file:   app/views/devise/sessions/new.html.haml
	new file:   app/views/devise/shared/_links.html.haml
	new file:   app/views/devise/unlocks/new.html.haml
	modified:   app/views/layouts/application.html.haml
	modified:   config/environments/development.rb
	new file:   config/initializers/devise.rb
	new file:   config/locales/devise.en.yml
	modified:   config/routes.rb
	new file:   db/migrate/20180211203709_devise_create_users.rb
	modified:   db/schema.rb
	new file:   spec/factories/users.rb
	new file:   spec/models/user_spec.rb

Commit:
decomposing main layout

Commit:
sign up confirmation added

Commit:
fix devise config for production

	modified:   config/environments/development.rb
	modified:   config/environments/production.rb
	modified:   config/initializers/devise.rb

Commit:
fix

Commit:
rubocop ref

Commit:
some views changed

Commit:
facebook sign in/sign up added

Commit:
namespacing changed

Commit:
refactoring

Commit:
draper implemented

Commit:
kaminari added

Commit:
filters implemented

Commit:
cancancan implenting

Commit:
ref

Commit:
making a guest user

Commit:
editing users addresses added

Commit:
refactoring relations

Commit:
tests ref

	modified:   app/models/order.rb
	modified:   app/models/review.rb
	new file:   db/migrate/20180221210234_drop_customers.rb
	modified:   db/schema.rb
	modified:   spec/factories/books.rb
	modified:   spec/factories/orders.rb
	deleted:    spec/factories/ratings.rb
	new file:   spec/factories/review.rb
	modified:   spec/models/author_spec.rb
	modified:   spec/models/book_spec.rb
	modified:   spec/models/category_spec.rb
	deleted:    spec/models/customer_spec.rb
	modified:   spec/models/image_spec.rb
	modified:   spec/models/order_spec.rb
	deleted:    spec/models/rating_spec.rb
	new file:   spec/models/review_spec.rb
	deleted:    spec/models/user_spec.rb

Commit:
removing remains of customer

Commit:
user privacy implemented

Commit:
tests ref

	modified:   app/models/address.rb
	modified:   config/environments/production.rb
	new file:   spec/factories/addresses.rb
	new file:   spec/models/address_spec.rb
	modified:   spec/models/order_spec.rb

Commit:
reviews implemented

	modified:   app/assets/javascripts/application.js
	new file:   app/assets/javascripts/review.js
	new file:   app/assets/stylesheets/custom.scss
	modified:   app/assets/stylesheets/style.scss
	modified:   app/controllers/application_controller.rb
	modified:   app/controllers/books_controller.rb
	new file:   app/controllers/reviews_controller.rb
	modified:   app/models/ability.rb
	modified:   app/models/address.rb
	modified:   app/models/review.rb
	modified:   app/models/user.rb
	modified:   app/views/books/show.html.haml
	modified:   config/locales/en.yml
	modified:   config/routes.rb
	new file:   db/migrate/20180223105238_change_reviews.rb
	modified:   db/schema.rb

Commit:
books characteristics implemented

Commit:
i18n ref

Commit:
raw functionality of cart added

	modified:   app/controllers/application_controller.rb
	new file:   app/controllers/order_items_controller.rb
	new file:   app/controllers/orders_controller.rb
	new file:   app/decorators/order_item_decorator.rb
	modified:   app/models/book.rb
	new file:   app/models/credit_card.rb
	modified:   app/models/order.rb
	new file:   app/models/order_item.rb
	modified:   app/models/user.rb
	new file:   app/views/order_items/index.html.haml
	modified:   app/views/partials/_book.html.haml
	modified:   app/views/partials/_header.html.haml
	modified:   config/routes.rb
	new file:   db/migrate/20180226122351_changing_orders.rb
	new file:   db/migrate/20180226123959_create_credit_cards.rb
	new file:   db/migrate/20180226131737_create_order_items.rb
	modified:   db/schema.rb
	new file:   spec/factories/credit_card.rb
	new file:   spec/models/credit_card_spec.rb
	new file:   spec/models/order_item_spec.rb
	modified:   spec/models/review_spec.rb
	new file:   spec/models/user_spec.rb

Commit:
coupons added

Commit:
ability to remove coupon from cart added

	modified:   app/controllers/coupons_controller.rb
	modified:   app/views/devise/registrations/edit.html.haml
	modified:   app/views/order_items/index.html.haml
	modified:   config/routes.rb

Commit:
make cart without signing in

Commit:
adding book from book page added

	modified:   app/assets/javascripts/application.js
	new file:   app/assets/javascripts/book_show.js
	modified:   app/assets/javascripts/review.js
	modified:   app/assets/javascripts/toggle_attr.js
	modified:   app/assets/javascripts/user_settings.js
	modified:   app/controllers/order_items_controller.rb
	modified:   app/decorators/book_decorator.rb
	modified:   app/models/address.rb
	modified:   app/views/books/show.html.haml
	modified:   app/views/home/index.html.haml
	modified:   app/views/order_items/index.html.haml
	modified:   app/views/partials/_address_form.html.haml
	modified:   config/routes.rb

Commit:
binding cart with user after successful singing in added

Commit:
stylize kaminari by bootstrap 3 and project theme

	modified:   app/assets/stylesheets/custom.scss
	modified:   app/controllers/categories_controller.rb
	modified:   app/views/categories/index.html.haml
	modified:   app/views/categories/show.html.haml
	new file:   app/views/kaminari/_first_page.html.haml
	new file:   app/views/kaminari/_gap.html.haml
	new file:   app/views/kaminari/_last_page.html.haml
	new file:   app/views/kaminari/_next_page.html.haml
	new file:   app/views/kaminari/_page.html.haml
	new file:   app/views/kaminari/_paginator.html.haml
	new file:   app/views/kaminari/_prev_page.html.haml
	modified:   app/views/partials/_book.html.haml

Commit:
some controller tests

Commit:
refactoring

Commit:
refactoring

	modified:   Gemfile
	modified:   app/assets/javascripts/book_show.js
	modified:   app/assets/javascripts/review.js
	modified:   app/assets/javascripts/toggle_attr.js
	modified:   app/assets/javascripts/user_settings.js
	modified:   app/controllers/addresses_controller.rb
	modified:   app/controllers/application_controller.rb
	modified:   app/controllers/categories_controller.rb
	modified:   app/controllers/home_controller.rb
	modified:   app/controllers/orders_controller.rb
	modified:   app/controllers/reviews_controller.rb
	modified:   app/controllers/users/omniauth_callbacks_controller.rb
	modified:   app/decorators/author_decorator.rb
	modified:   app/decorators/book_decorator.rb
	modified:   app/decorators/order_decorator.rb
	modified:   app/decorators/order_item_decorator.rb
	modified:   app/helpers/application_helper.rb
	modified:   app/models/address.rb
	modified:   app/models/book.rb
	modified:   app/models/order.rb
	modified:   app/models/user.rb
	modified:   app/views/books/_make_review.html.haml
	modified:   app/views/books/show.html.haml
	modified:   app/views/home/index.html.haml
	modified:   app/views/order_items/index.html.haml
	modified:   app/views/partials/_book.html.haml
	modified:   config/environments/test.rb
	modified:   config/initializers/kaminari_config.rb
	modified:   config/initializers/simple_form.rb
	modified:   config/initializers/simple_form_bootstrap.rb
	modified:   config/routes.rb
	modified:   db/migrate/20180220191136_changing_references_between_books_and_authors.rb
	modified:   db/schema.rb
	modified:   db/seeds.rb
	modified:   spec/controllers/addresses_controller_spec.rb
	modified:   spec/controllers/books_controller_spec.rb
	modified:   spec/controllers/categories_controller_spec.rb
	modified:   spec/factories/credit_card.rb
	modified:   spec/models/book_spec.rb
	modified:   spec/models/order_item_spec.rb
	modified:   spec/models/order_spec.rb

Commit:
Merge branch 'dev' of github.com:6aBblKuH/bookstore into dev

kirillshevch commented on app/controllers/application_controller.rb:
```diff
@@ -1,3 +1,26 @@
+  end
+
+  def current_order
+    Order.find_by_id(session[:order_id]) || current_user&.orders&.in_progress&.last || new_session_order
```
`@current_order ||= (Order.find_by(id: session[:order_id]) || current_user.active_order || new_session_order).decorate`

Move `&.orders&.in_progress&.last` to scope `active_order`

kirillshevch commented on app/models/book.rb:
```diff
@@ -0,0 +1,28 @@
+  has_many :images, dependent: :destroy
+  has_many :order_items
+  belongs_to :category
+  has_and_belongs_to_many :author
```
has_and_belongs_to_many :authors

kirillshevch commented on app/controllers/categories_controller.rb:
```diff
@@ -0,0 +1,17 @@
+  load_resource
+  before_action :set_filter
+  def index
+    @books = Book.public_send(@filter).page(params[:page])
```
move to scope to avoid dup with method show`.public_send(@filter).page(params[:page])`

kirillshevch commented on app/controllers/order_items_controller.rb:
```diff
@@ -0,0 +1,32 @@
+  load_and_authorize_resource
+
+  def index
+    return redirect_to root_path, alert: 'You`ve not any books in cart' unless @order.order_items.any?
```
`'You`ve not any books in cart'` to i18n

kirillshevch commented on app/controllers/order_items_controller.rb:
```diff
@@ -0,0 +1,32 @@
+  end
+
+  def destroy
+    OrderItem.destroy(params[:id])
```
@order_item.destroy

kirillshevch commented on app/controllers/orders_controller.rb:
```diff
@@ -0,0 +1,2 @@
+class Orders < ApplicationController
```
OrdersController

kirillshevch commented on app/controllers/books_controller.rb:
```diff
@@ -0,0 +1,9 @@
+
+  def show
+    @book = @book.decorate
+    @reviews = Review.where(book: @book).confirmed
```
@reviews = @book.reviews.confirmed

kirillshevch commented on app/controllers/reviews_controller.rb:
```diff
@@ -0,0 +1,13 @@
+  def create
+    @review = current_user.reviews.create(review_params)
+    @review.errors.any? ? flash[:alert] = t('review.failure.create') : flash[:notice] = t('review.successful.create')
+    redirect_to book_path(review_params[:book_id])
```
redirect_to @book

kirillshevch commented on app/controllers/reviews_controller.rb:
```diff
@@ -0,0 +1,13 @@
+class ReviewsController < ApplicationController
+  def create
```
load_and_authorize_resource

kirillshevch commented on config/routes.rb:
```diff
@@ -1,3 +1,18 @@
+  mount RailsAdmin::Engine => '/admin', as: 'rails_admin'
+  root 'home#index'
+  resources :categories, only: %i[index show]
+  resources :books, only: :show
```
```ruby
resources :books, only: :show do
  resources :reviews, only: :create
end
```

kirillshevch commented on app/helpers/application_helper.rb:
```diff
@@ -1,2 +1,9 @@
+  end
+
+  def qunatity_book_in_category(id)
+    Book.where(category_id: id).length
```
Book.where(category_id: id).count

kirillshevch commented on app/models/credit_card.rb:
```diff
@@ -0,0 +1,4 @@
+class CreditCard < ApplicationRecord
+  has_many :orders
```
belongs_to :user

Commit:
refactoring after review & tests

Commit:
few feature tests

Commit:
checkout orders added

Commit:
checkout upgrade

Commit:
checkout confirm

Commit:
checkout complete

	modified:   app/controllers/order_steps_controller.rb
	modified:   app/controllers/orders_controller.rb
	modified:   app/forms/order_steps_form.rb
	new file:   app/mailers/order_mailer.rb
	modified:   app/models/ability.rb
	modified:   app/models/credit_card.rb
	modified:   app/models/delivery.rb
	modified:   app/models/order.rb
	modified:   app/models/user.rb
	modified:   app/views/devise/registrations/edit.html.haml
	new file:   app/views/layouts/mailer.html.haml
	new file:   app/views/order_mailer/confirm_order.html.haml
	modified:   app/views/order_steps/address.html.haml
	new file:   app/views/order_steps/complete.html.haml
	modified:   config/environments/development.rb
	modified:   config/environments/production.rb
	modified:   config/environments/test.rb
	modified:   config/routes.rb
	new file:   db/migrate/20180316120551_change_orders.rb
	modified:   db/schema.rb

Commit:
user orders list in account added

	modified:   app/controllers/orders_controller.rb
	modified:   app/controllers/reviews_controller.rb
	modified:   app/models/ability.rb
	modified:   app/models/order.rb
	new file:   app/views/orders/index.html.haml
	modified:   app/views/partials/_footer.html.haml
	modified:   app/views/partials/_header.html.haml
	modified:   config/routes.rb

Commit:
user orders show

kirillshevch commented on app/controllers/categories_controller.rb:
```diff
@@ -0,0 +1,19 @@
+class CategoriesController < ApplicationController
+  load_resource
```
load_and_authorize_resource

kirillshevch commented on app/controllers/coupons_controller.rb:
```diff
@@ -0,0 +1,21 @@
+  end
+
+  def destroy
+    Coupon.find(params[:id]).update(order_id: nil)
```
ability?

kirillshevch commented on app/models/ability.rb:
```diff
@@ -0,0 +1,19 @@
+    user ||= User.new
+
+    can :read, :all
+    can :manage, OrderItem
```
```ruby
can :manage, OrderItem, order_id: current_order.id
```

kirillshevch commented on db/migrate/20180227134527_create_coupons.rb:
```diff
@@ -0,0 +1,9 @@
+class CreateCoupons < ActiveRecord::Migration[5.1]
+  def change
+    create_table :coupons do |t|\
```
\?

kirillshevch commented on db/migrate/20180310123548_make_references_for_credit_card_and_order.rb:
```diff
@@ -0,0 +1,6 @@
+class MakeReferencesForCreditCardAndOrder < ActiveRecord::Migration[5.1]
+  def change
+    add_reference :credit_cards, :user, index: true
```
add_reference :credit_cards, :user, index: true, foreign_key: true
